htlcnotifier/interface.go

```diff
@@ -0,0 +1,21 @@
+package htlcnotifier
+
+// Notifier represents a place through which HTLC events can be piped through.
+//
+// Concrete implementations of Notifier should be non-blocking.
+type Notifier interface {
+	// NotifyForwardEvent is called when we successfully forward a HTLC.
+	NotifyForwardingEvent(f ForwardingEvent)
+
+	// NotifyLinkFailEvent is called when we fail to forward a HTLC over our
+	// outgoing link.
+	NotifyLinkFailEvent(reason FailReason, f ForwardingEvent)
+
+	// NotifySettleEvent is called when a HTLC we previously successfully
+	// forwarded is settled.
+	NotifySettleEvent(f ForwardingEvent)
+
+	// NotifyForwardingFailEvent is called when a HTLC we previously
+	// successfully forwarded fails down the line.
+	NotifyForwardingFailEvent(failType ForwardFailType, f ForwardingEvent)
```

Comment:
👍 updated in the more recent WIP draft, cleaned these up into individual event types with a common identifier struct (ID, to be renamed to something better) rather than trying to tack failures onto `ForwardingEvent` when needed.